Implement --source command-line argument.

gergoerdi opened the pull request:
When passed, instead of starting a new subprocess with a changed environment,
the intended new value of $PATH is printed out in a format suitable to
be piped into eval:

```
eval `ghc-env --source`
```

will result in the current shell's environment being changed to include
the selected GHC version in $PATH.

sol commented:
Hey, this looks nice.  We haven't had travis enabled for this repo, but I just did that and pushed it to `master`.

Can you do the following?
- rebase on `master` and make sure that travis is green
- mentioning this in the readme would be nice, too

More on the bike shedding side of things, would `--env` or `--path` be a better fit for the user interface?  I'm not very opinionated either way, just want to throw in two more options.  I'm fine with what ever is your preference.

gergoerdi committed:
Implement --env command-line argument.

When passed, instead of starting a new subprocess with a changed environment,
the intended new value of $PATH is printed out in a format suitable to
be piped into eval:

    eval `ghc-env --env`

will result in the current shell's environment being changed to include
the selected GHC version in $PATH.

gergoerdi committed:
Add --env flag to README

gergoerdi committed:
-Wall cleanup

gergoerdi committed:
Update tests for --env flag

gergoerdi commented:
You'll have to nudge Travis a bit:

> ```
> Err http://ppa.launchpad.net/hvr/ghc/ubuntu/ precise/main cabal-install-1.22 amd64 1.22.6.0-1~precise
>   Unable to connect to ppa.launchpad.net:http:
> ```

```
```

sol commented:
Passing now, thanks for fixing the tests and the readme update.
